src/hooks/useFormErrors.ts

```diff
@@ -0,0 +1,62 @@
+import { useState, useCallback, useMemo } from 'react';
+import produce from 'immer';
+
+export interface IFormErrors {
+    // Get the error message for a field name, if any.
+    get(field: string): string | undefined;
+
+    // Set an error message for a field name.
+    set(field: string, message: string): void;
+
+    // Remove an existing error for a field name.
+    remove(field: string): void;
+
+    // Check if there are any errors.
+    some(): boolean;
+}
+
+export const useFormErrors = (): IFormErrors => {
+    const [errors, setErrors] = useState<Record<string, string>>({});
+
+    const get = useCallback(
+        (field: string): string | undefined => errors[field],
+        [errors]
+    );
+
+    const set = useCallback(
+        (field: string, message: string): void => {
+            setErrors(
+                produce(draft => {
+                    draft[field] = message;
+                })
+            );
+        },
+        [setErrors]
+    );
+
+    const remove = useCallback(
+        (field: string): void => {
+            setErrors(
+                produce(draft => {
+                    delete draft[field];
+                })
+            );
+        },
+        [setErrors]
+    );
+
+    const some = useCallback(
+        (): boolean => Object.values(errors).some(Boolean),
+        [errors]
+    );
+
+    return useMemo(
```

Comment:
What is the value of using useMemo here when all of the references are stabilized by useCallback? Seems to me we're only adding overhead with this.

Reply:
The idea was that the return value from this hook (`errors`) should be a stable reference, so that it can be passed to optimized components directly, and used in `useEffect` dependency arrays (instead of passing each fn separately).

Reply:
But are we using it in any use effects? Or is this a premature optimization? 

Would it make sense to put the entire errors object in a useEffect? My hunch is that you would rarely all of these methods inside the useEffect? If you use `errors.get()` or `errors.some` those would still be stable references or am I missing something here?

Reply:
True, not needed yet. I can remove the last `useMemo` and instead expand the function names to guide their usage.